common/hex.c

```diff
@@ -64,3 +66,44 @@ hex_encode (const unsigned char *data,
 	result[o] = 0;
 	return result;
 }
+
+unsigned char *
+hex_decode (const char *hex,
+            size_t *bin_len)
+{
+	size_t i, j;
+	unsigned long val;
+	unsigned char *bin;
+	char hex2[3] = { 0 };
+        size_t hex_len = strlen (hex);
+
+	bin = malloc (hex_len / 2);
+	if (bin == NULL)
+		return NULL;
+
+        for (i = j = 0; i < hex_len;) {
+		if (hex[i] == ':') {
+			++i;
+			continue;
```

Comment:
This is too lax as it accepts colons in an invalid position, e.g., `00::::::::11`.